client/object_put.go

```diff
@@ -168,28 +175,29 @@ func (x *ObjectWriter) WritePayloadChunk(chunk []byte) bool {
 		// the allocated buffer is filled, or when the last chunk is received.
 		// It is mentally assumed that allocating and filling the buffer is better than
 		// synchronous sending, but this needs to be tested.
+		writtenBytes += len(chunk[:ln])
 		x.partChunk.SetChunk(chunk[:ln])
 		x.req.SetVerificationHeader(nil)
 
 		x.err = signServiceMessage(x.signer, &x.req)
 		if x.err != nil {
 			x.err = fmt.Errorf("sign message: %w", x.err)
-			return false
+			return writtenBytes, x.err
 		}
 
 		x.err = x.stream.Write(&x.req)
 		if x.err != nil {
-			return false
+			return writtenBytes, x.err
```

Comment:
Failed to send and returning `writtenBytes`? Although here it can be unobvious what amount of bytes was sent or not (if we're not examining `x.err`).

Reply:
Moved `writtenBytes += len(chunk[:ln])` to the end of for loop, _writtenBytes_ should be correct now.

It it better to return 0, instead of _writtenBytes_ on error in this and upper case?

Reply:
We should return the number of bytes written.

Reply:
Ok, moving `writtenBytes++` should solve this and problem above